src/main/scala/com/exasol/cloudetl/kafka/consumer/KafkaRecordConsumer.scala

```diff
@@ -57,11 +58,64 @@ class KafkaRecordConsumer(
         )
       } while (shouldContinue(recordOffset, recordCount, totalRecordCount))
     } catch {
+      case exception: IllegalStateException =>
+        throw new KafkaConnectorException(
+          ExaError
+            .messageBuilder("E-KCE-20")
+            .message("Error consuming Kafka topic {{TOPIC}} data. ", topic)
+            .message("Consumer is not subscribed to any topic or assigned any partition.")
+            .mitigation("Please check that the Kafka topic is available and valid.")
+            .toString(),
+          exception
+        )
+      case exception: InvalidTopicException =>
+        throw new KafkaConnectorException(
+          ExaError
+            .messageBuilder("E-KCE-21")
+            .message(generalErrorMessage(), topic)
+            .message("Provided topic is not valid.")
+            .mitigation("Please check that the Kafka topic is available and valid.")
+            .toString(),
+          exception
+        )
+      case exception: TimeoutException =>
+        throw new KafkaConnectorException(
+          ExaError
+            .messageBuilder("E-KCE-22")
+            .message(generalErrorMessage(), topic)
+            .message("Timeout trying to connect to Kafka brokers.")
+            .mitigation(
+              "Please ensure that there is network connection between Kafka brokers and Exasol datanode." +
+                "Similarly check that Kafka advertised listeners are reachable from Exasol cluster."
+            )
+            .toString(),
+          exception
+        )
+      case exception: AuthorizationException =>
+        throw new KafkaConnectorException(
+          ExaError
+            .messageBuilder("E-KCE-23")
+            .message(generalErrorMessage(), topic)
+            .message("Consumer or consumer group is not allowed to read given topic. Cause: " + exception.getMessage())
+            .mitigation("Please make sure that topic is readable by the this consumer or consumer groups")
```

Comment:
```suggestion
            .mitigation("Please make sure that the topic is readable by this consumer or consumer groups")
```

Reply:
You could also remove passive here: `that consumer or consumer groups can read...`